app/models/budget.rb

```diff
@@ -47,13 +47,32 @@ def period_end_date
   end
 
   def iati_type
-    IATI_TYPES.fetch("original")
+    return IATI_TYPES.fetch("original") if original?
+
+    IATI_TYPES.fetch("revised")
   end
 
   def iati_status
     IATI_STATUSES.fetch("indicative")
   end
 
+  def original_revision
+    audits.find_by(action: "create").revision
+  end
+
+  def any_update_audits?
+    audits.updates.any?
+  end
+
+  private def original?
```

Comment:
I'm not sure whether I'm particularly happy with this..... it does work though!

It will raise and error if you did something like `Budget.new.iati_type`, which I think is fine?

Reply:
Okay, I still think the modeling here is weird.

If a Budget has revisions, I don't think a Budget should have an `iati_type`, I only think a revision should have an `iati_type`

I've attempted to address this [here](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/commit/f4c88521b8585fd5101bffe07794c2dfc2fd087b) 

The result is ultimately the same, it's whatever makes most sense to everyone.

Reply:
okay, I've taken inspiration from `paper_trail` which has a `live?` method to check whether a record is live or a revision. I'm fairly happy now